homeassistant/components/fritzbox/config_flow.py

```diff
@@ -110,12 +114,20 @@ async def async_step_ssdp(self, user_input):
         host = urlparse(user_input[ATTR_SSDP_LOCATION]).hostname
         self.context[CONF_HOST] = host
 
+        uuid = user_input.get(ATTR_UPNP_UDN)
+        if uuid:
+            if uuid.startswith("uuid:"):
+                uuid = uuid[5:]
+            await self.async_set_unique_id(uuid)
+
         for progress in self._async_in_progress():
             if progress.get("context", {}).get(CONF_HOST) == host:
                 return self.async_abort(reason="already_in_progress")
 
         for entry in self.hass.config_entries.async_entries(DOMAIN):
             if entry.data[CONF_HOST] == host:
+                if uuid and entry.unique_id != uuid:
```

Comment:
Only do this if unique ID is not set. If unique ID is different, it's a different box!
```suggestion
                if uuid and not entry.unique_id:
```